docs/validators.md

```diff
@@ -0,0 +1,21 @@
+Validators
+==========
+
+Validators allow you to validate or transform command arguments just before execution.
+
+Knack supports both command-level and argument-level validators.
+
+If a command has a command-level validator, then any argument-level validators that would ordinarily be applied are ignored.
+
+i.e. A command can have at most command validator or many argument level validators.
+
+**Command-level Validators**
+
+Command-level validators can operate on any arguments on the command. This is useful when you want a validator to make use of more than a single argument.
```

Comment:
An argument validator can always use multiple arguments. I think a more accurate statement here would be "This is useful when you need to control the validation sequence."

The best example I can give is when you want to use the "get_location_from_resource_group" validator. This would require a command validator because the location needs to be resolved first if other validators might need to make use of the computed location.